src/mixins/raster-layer-mesh2d-mixin.js

```diff
@@ -0,0 +1,302 @@
+      layerName,
+      lastFilteredSize
+    )
+
+    _vega_property_output_state = materializePropDescriptors(
+      raster_layer_context,
+      prop_descriptors,
+      state
+    )
+    const {
+      sql_parser_transforms,
+      vega_data_formats,
+      vega_transforms,
+      vega_scales,
+      mark_properties
+    } = _vega_property_output_state.flatten()
+
+    assert(vega_data_formats.length <= 1)
+
+    if (typeof filter === "string" && filter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: filter
+      })
+    }
+
+    const post_filter_transform = create_post_filter_transform(
+      state.postFilters
+    )
+
+    if (post_filter_transform) {
+      sql_parser_transforms.push(post_filter_transform)
+    }
+
+    if (typeof globalFilter === "string" && globalFilter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: globalFilter
+      })
+    }
+
+    const data = [
+      {
+        name: layerName,
+        ...(vega_data_formats.length ? vega_data_formats[0] : {}),
+        sql: parser.writeSQL({
+          type: "root",
+          source: table,
+          transform: sql_parser_transforms
+        }),
+        enableHitTesting: state.enableHitTesting
+      },
+      ...vega_transforms
+    ]
+
+    const marks = [
+      {
+        type: Mesh2dConfigDefinitionObject.key,
+        from: {
+          data: layerName
+        },
+        properties: Object.assign({}, ...mark_properties)
+      }
+    ]
+
+    return { data, scales: vega_scales, marks }
+  }
+
+  function isCrossFilterObject(obj) {
+    if (
+      Object.hasOwn(obj, "type") &&
+      Object.hasOwn(obj, "getCrossfilter") &&
+      Object.hasOwn(obj, "filter")
+    ) {
+      // is this safe to assume that if these properties exist, then the object
+      // is a crossfilter-related object?
+      return true
+    }
+    return false
+  }
+
+  function createCrossfilterDimensionProxy(range) {
+    assert(Array.isArray(range))
+    assert(range.length === 2)
+    let range_ = [...range]
+    return {
+      type: "crossfilter_dimension_proxy",
+      getFilter: () => [range_],
+      filter: new_range => {
+        assert(Array.isArray(new_range))
+        assert(new_range.length === 2)
+        range_ = new_range
+      }
+    }
+  }
+
+  _layer.xDim = createRasterLayerGetterSetter(_layer, null, new_val => {
+    if (isCrossFilterObject(new_val)) {
+      return new_val
+    }
+    return createCrossfilterDimensionProxy(new_val)
+  })
+  _layer.yDim = createRasterLayerGetterSetter(_layer, null, new_val => {
+    if (isCrossFilterObject(new_val)) {
+      return new_val
+    }
+    return createCrossfilterDimensionProxy(new_val)
+  })
+
+  let _vega = null
+  const _cf = null
+  _layer.crossfilter = createRasterLayerGetterSetter(_layer, _cf)
+
+  // eslint-disable-next-line no-unused-vars
+  _layer._genVega = function(chart, layerName, group, query) {
+    const realLayerName = layerName
```

Comment:
Why rename the param here?

Reply:
some copy-pasta carry-over originally from the raster points mixin variant. It can certainly go.